contexts/DelegationContext.tsx

```diff
@@ -0,0 +1,292 @@
+import { zeroAddress } from "helpers";
+import {
+  useAcceptReceivedRequestToBeDelegate,
+  useCancelSentRequestToBeDelegate,
+  useContractsContext,
+  useDelegateToStaker,
+  useDelegatorSetEventsForDelegate,
+  useDelegatorSetEventsForDelegator,
+  useIgnoredRequestToBeDelegateAddresses,
+  useIgnoreReceivedRequestToBeDelegate,
+  usePanelContext,
+  useReceivedRequestsToBeDelegate,
+  useSendRequestToBeDelegate,
+  useSentRequestsToBeDelegate,
+  useStakingContext,
+  useTerminateRelationshipWithDelegate,
+  useTerminateRelationshipWithDelegator,
+  useUserContext,
+  useVoterFromDelegate,
+} from "hooks";
+import { createContext, ReactNode } from "react";
+import { DelegationEventT, DelegationStatusT } from "types";
+export interface DelegationContextState {
+  getDelegationStatus: () => DelegationStatusT;
+  getPendingReceivedRequestsToBeDelegate: () => DelegationEventT[];
+  getHasPendingReceivedRequestsToBeDelegate: () => boolean;
+  getPendingSentRequestsToBeDelegate: () => DelegationEventT[];
+  getHasPendingSentRequestsToBeDelegate: () => boolean;
+  getDelegateAddress: () => string | undefined;
+  getDelegatorAddress: () => string | undefined;
+  sendRequestToBeDelegate: (delegateAddress: string) => void;
```

Comment:
The actions the user can do are:

* send be to be delegate — you are asking another wallet to be your delegate. This calls `setDelegate(otherWallet)`
* cancel sent request to be delegate — undoes the above by calling `setDelegate(zeroAddress)
* accept received request to be delegate — completes the two-way relationship from the other wallet by calling `setDelegator(walletRequestCameFrom)`
* ignore request to be delegate — if someone you don't want requests you to be their delegate, there's no way to stop them through the contracts. Instead we add the unwanted requester's address to a list of ignored addresses in local storage and don't show pending requests from there.